external/llvm-project/mlir/lib/Conversion/GPUToROCDL/LowerGpuOpsToROCDLOps.cpp

```diff
@@ -875,6 +875,17 @@ struct MFMAOpLowering : ConvertToLLVMPattern {
             op, adaptor.destC().getType(),
             ValueRange{adaptor.sourceA(), adaptor.sourceB(), adaptor.destC(),
                        immValues[0], immValues[1], immValues[2]});
+    } else if (mfmaInstr.endswith("i8")) {
+      if (mfmaInstr == "mfma_i32_32x32x8i8")
+        rewriter.replaceOpWithNewOp<ROCDL::mfma_i32_32x32x8i8>(
+            op, adaptor.destC().getType(),
+            ValueRange{adaptor.sourceA(), adaptor.sourceB(), adaptor.destC(),
+                       immValues[0], immValues[1], immValues[2]});
+      else if (mfmaInstr == "mfma_i32_16x16x16i8")
+        rewriter.replaceOpWithNewOp<ROCDL::mfma_i32_16x16x16i8>(
```

Comment:
We sure those're the only ones we'll need? Not all our kernels have k reductions?

Reply:
> We sure those're the only ones we'll need? Not all our kernels have k reductions?

No I'm not sure but it should be easy enough to add new ones if necessary. Right now I just expose the absolutely minimal. In case we figure later that more needs to be done, I will notice that.